### PR TITLE
Add src layout structure

### LINKED ISSUE #1: Project layout
This project should have a [src-layout](https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/). The package name should have according to [pep 8](https://peps.python.org/pep-0008/#package-and-module-names) the name `cobbler_tftp`.

We do want to have a separate docs folder that is located in the project root.

Package data support is managed by setuptools and we want to use a [subdirectory for it](https://setuptools.pypa.io/en/latest/userguide/datafiles.html#subdirectory-for-data-files). Due to the fact that we want to stay compatible with Python 3.6 to run successfully on enterprise distributions like SLES (setuptools 44.1.1 according to SCC in SP5), we need to add a dependency on [importlib-resources](https://pypi.org/project/importlib-resources/).

For testing we want to utilize [pytest](https://docs.pytest.org/en/7.0.x/) with version 7.0.x, as this is the last Python 3.6 compatible one. For mocking we will use [pytest-mock](https://pypi.org/project/pytest-mock/3.6.1/) in version 3.6.1 as this is the last Python 3.6 compatible one. The tests will be located in the folder `tests` which is to be found in the project root.

### PR DESCRIPTION
**Pull request by ByteOtter**
Fixes #1, #2 

Introduce src-layout to the project alongside the necessary configuration files.

### CODE REVIEW

**Review by SchoolGuy — CHANGES_REQUESTED** [review]
`setup.cfg` has some old leftovers of the PoC. Let's change that please.

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+long_description_content_type = text/markdown
+keywords = cobbler, tftp
+license = GPL v2.0 License
+classifiers =
+  Natural Language :: English
+  Operating System :: OS Independent
+  Programming Language :: Python :: 3
+
+[options]
+zip_safe = Frue
```
Comment (SchoolGuy) [inline setup.cfg:17]:
```suggestion
zip_safe = True
```

or 

```suggestion
zip_safe = False
```

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,67 @@
+[metadata]
+name = cobbler_tftp
+author = Christopher Hock
+author_email = christopher.hock@suse.com
```
Comment (SchoolGuy) [inline setup.cfg:4]:
```suggestion
author = Team Cobbler
author_email = cobbler.project@gmail.com
```

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  packaging~=23.0
+  pathspec~=0.11.0
+  platformdirs~=3.1.0
+  pluggy~=1.0.0
+  py~=1.11.0
+  python-daemon~=3.0.1
+  PyYAML~=6.0
+  tomli~=2.0.1
+  typing_extensions~=4.5.0
+  importlib-resources~=5.12.0
```
Comment (SchoolGuy) [inline setup.cfg:47]: There are some linters hiding in there.

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,67 @@
+[metadata]
+name = cobbler_tftp
+author = Christopher Hock
+author_email = christopher.hock@suse.com
+url = https://github.com/ByteOtter/cobbler-tftp
```
[the remark this answers is not in the document]
Comment (ByteOtter) [reply setup.cfg]: fixed 👍🏼

File: setup.cfg (hunk 1 above)
Comment (ByteOtter) [reply setup.cfg]: opted for true for now 👍🏼

File: setup.cfg (hunk 3 above)
Comment (ByteOtter) [reply setup.cfg]: removed black from there and moved the extra dependencies into their own `[options.extras_require]` block

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  attrs~=22.2.0
+  click>=8.0.0
+  exceptiongroup~=1.1.0
+  fbtftp~=0.5
+  iniconfig~=2.0.0
+  Jinja2~=3.1.2
+  lockfile~=0.12.2
+  MarkupSafe~=2.1.2
+  mypy-extensions~=1.0.0
+  packaging~=23.0
```
Comment (SchoolGuy) [inline setup.cfg:38]: This package requires Python 3.7 or greater and I doubt it is required for the runtime of the project.

File: setup.cfg [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  pip~=23.0.1
+  attrs~=22.2.0
+  click>=8.0.0
+  exceptiongroup~=1.1.0
+  fbtftp~=0.5
+  iniconfig~=2.0.0
+  Jinja2~=3.1.2
+  lockfile~=0.12.2
+  MarkupSafe~=2.1.2
+  mypy-extensions~=1.0.0
```
Comment (SchoolGuy) [inline setup.cfg:37]: This is just for typing. Are we sure we need that? Seems to me that with `pyright` this dependency is not required anymore.

File: setup.cfg [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  =src
+packages = find:
+include_package_data = True
+setup_requires =
+  setuptools<=44.11.0
+  setuptools-scm~=7.1.0
+  wheel~=0.38.4
+install_requires =
+  python_version>=3.6
+  pip~=23.0.1
```
Comment (SchoolGuy) [inline setup.cfg:28]: That for sure is not needed for running a tftp server IMHO. I can also not figure out why we need exactly this version for developing this project.

File: setup.cfg [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  fbtftp~=0.5
+  iniconfig~=2.0.0
+  Jinja2~=3.1.2
+  lockfile~=0.12.2
+  MarkupSafe~=2.1.2
+  mypy-extensions~=1.0.0
+  packaging~=23.0
+  pathspec~=0.11.0
+  platformdirs~=3.1.0
+  pluggy~=1.0.0
```
Comment (SchoolGuy) [inline setup.cfg:41]: I know that this is a dependency of `pytest`. I don't believe we will need it at this early stage of the project. If we need it please feel free to correct me.

File: setup.cfg [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+  py~=1.11.0
```
Comment (SchoolGuy) [inline setup.cfg:42]: This library is in [maintenance mode](https://pypi.org/project/py/). I am not aware why we would need that?

File: setup.cfg [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+  python-daemon~=3.0.1
+  PyYAML~=6.0
+  tomli~=2.0.1
+  typing_extensions~=4.5.0
```
Comment (SchoolGuy) [inline setup.cfg:46]: Is there a particular reason why we need this package?

File: setup.cfg [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+  MarkupSafe~=2.1.2
+  mypy-extensions~=1.0.0
+  packaging~=23.0
+  pathspec~=0.11.0
+  platformdirs~=3.1.0
+  pluggy~=1.0.0
+  py~=1.11.0
+  python-daemon~=3.0.1
+  PyYAML~=6.0
+  tomli~=2.0.1
```
Comment (SchoolGuy) [inline setup.cfg:45]: Do we need this as well? The config is in YAML and I am not aware of any TOML files we need to parse atm.

File: setup.cfg [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+doc =
+  Sphinx~=6.1.3
+  sphinx-rtd-theme~=1.2.0
+  sphinxcontrib-applehelp~=1.0.4
+  sphinxcontrib-devhelp~=1.0.2
+  sphinxcontrib-htmlhelp~=2.0.1
+  sphinxcontrib-jquery~=2.0.0
+  sphinxcontrib-jsmath~=1.0.1
+  sphinxcontrib-qthelp~=1.0.3
+  sphinxcontrib-serializinghtml~=1.1.5
```
Comment (SchoolGuy) [inline setup.cfg:65]: Those are dependencies of Sphinx itself I guess. Explicitly we don't need them afaik. Let's try to get rid of them.

File: setup.cfg [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+  Programming Language :: Python :: 3
+
+[options]
+zip_safe = True
+package_dir=
+  =src
+packages = find:
+include_package_data = True
+setup_requires =
+  setuptools<=44.11.0
```
Comment (SchoolGuy) [inline setup.cfg:23]: So I believe that we don't need something that is smaller or equal to 44.11.0. I believe that atm we should be able to install this project also with more modern versions of `setuptools`. So I believe we want this if I am not mistaken:

```suggestion
  setuptools>=44.11.0
```

**Review by SchoolGuy — CHANGES_REQUESTED** [review]
I am sorry but the dependencies we name are not in a shape I feel confident where we can already merge this PR.

If you desire that we version lock all of our dependencies explicitly (it is a valid strategy after all) then I would like us to use [Poetry](https://python-poetry.org/). However, this is not needed for this project. Well-thought version constraints should be enough.

I am sorry if I have overlooked other dependencies that this project doesn't need but with the questions I have provided in the individual comments, we should already have a much leaner dependency tree.

File: setup.cfg (hunk 11 above)
Comment (ByteOtter) [reply setup.cfg:45]: Most of these requirements like `tomli` are requirements of requirements. In this case this is a requirements of `setuptools-scm`. I'm currently looking up if we need to declare them and will clean them up accordingly

**Comment by ByteOtter** [comment]
> I am sorry but the dependencies we name are not in a shape I feel confident where we can already merge this PR.
> 
> If you desire that we version lock all of our dependencies explicitly (it is a valid strategy after all) then I would like us to use [Poetry](https://python-poetry.org/). However, this is not needed for this project. Well-thought version constraints should be enough.
> 
> I am sorry if I have overlooked other dependencies that this project doesn't need but with the questions I have provided in the individual comments, we should already have a much leaner dependency tree.

Alright, newest update should fix all these concerns. I changed all dependency versions to their latest release which still supported Python 3.6 and removed all which are not explicitly needed.

File: setup.cfg [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+package_dir=
+  =src
+packages = find:
+include_package_data = True
+setup_requires =
+  setuptools>=44.11.0
+  setuptools-scm~=7.1.0
+  wheel>=0.37.1
+install_requires =
+  python_version>=3.6
```
Comment (SchoolGuy) [inline setup.cfg:27]: Is this dependency here on purpose?

I see it mapping to this project: https://gitlab.com/halfak/python_version

File: setup.cfg [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+  pip>=21.3.1
```
Comment (SchoolGuy) [inline setup.cfg:28]: `pip` is not needed to run the project. This project doesn't integrate with `pip` and thus we can maybe open the `dev_requires` section for that but please let's remove that from the install_requires.

**Review by SchoolGuy — COMMENTED** [review]
Looking much better! Two questions/open points are left.

File: setup.cfg (hunk 15 above)
Comment (ByteOtter) [reply setup.cfg:28]: moved it to a new `dev_requires` section

File: setup.cfg (hunk 14 above)
Comment (ByteOtter) [reply setup.cfg:27]: It shouldn't go there. I may have misinterpreted the (`setup.cfg Example`)[https://web.archive.org/web/20200118161529/https://setuptools.readthedocs.io/en/latest/setuptools.html#using-a-src-layout] in the old setuptools documentation and was led to believe I need to specify the exact python version in the `install_requires` section.

However, upon closer inspection it looks like I need to do so in the `classifiers` section with
```cfg
Programming Language :: Python :: 3.6
```

**Review by SchoolGuy — APPROVED** [review]
LGTM

**Pull request merged**